aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
-.. include:: /../common/stub-topic.txt
+Formatting Response Data
+========================
 
-|stub-icon| Formatting
-======================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET Core MVC has built-in support for formatting response data, using fixed formats or in response to client specifications.
 
-.. _issue: https://github.com/aspnet/Docs/issues/122
+.. contents:: Sections
+    :local:
+    :depth: 1
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/master/mvc/models/formatting/sample>`_.
+
+Format-Specific Action Results
+------------------------------
+
+Some ActionResult options are specific to a particular format, such as JsonResult and ContentResult (note strings are simply returned as text). Actions can return specific results that are always formatted in a particular manner. For example, returning a ``JsonResult`` will return JSON-formatted data, regardless of client preferences. Likewise, returning a ``ContentResult`` will return plain text formatted string data.
+
+To return data using a specific result type, assuming you're inheriting from the base ``Controller`` class, it's recommended that you use the built-in helper methods ``Json`` and ``Content``. These methods check to see if the data being formatted implements ``IDisposable`` and register the object for disposal if required. Your action method should return either the specific result type (for instance, ``JsonResult``) or ``IActionResult``.
```

Comment:
Oops

... the `object` will be serialized using the appropriate `IOutputFormatter` implementation.

Are you looking for a list of provided `IActionResult` implementations, what they do, and the corresponding `Controller` method (if any)? I'd hope "what they do" is fairly obvious from their names and doc comments. So I'm thinking it would be a large table with little value.

/cc @danroth27